src/panels/logbook/ha-logbook.ts

```diff
@@ -254,6 +265,126 @@ class HaLogbook extends LitElement {
     this._savedScrollPos = (e.target as HTMLDivElement).scrollTop;
   }
 
+  private _formatEventBy(item: LogbookEntry, seenEntities: string[]) {
+    if (item.context_event_type === "call_service") {
+      return `${this.hass.localize("ui.components.logbook.from_service")} ${
+        item.context_domain
+      }.${item.context_service}`;
+    }
+    if (
+      item.context_event_type === "automation_triggered" &&
+      item.context_entity_id
+    ) {
+      if (seenEntities.includes(item.context_entity_id)) {
+        return ``;
+      }
+      seenEntities.push(item.context_entity_id);
+      return html`${this.hass.localize("ui.components.logbook.from_automation")}
+      ${this._renderEntity(item.context_entity_id, item.context_name)}`;
+    }
+    if (item.context_name) {
+      return `${this.hass.localize("ui.components.logbook.from")} ${
+        item.context_name
+      }`;
+    }
+    if (item.context_event_type === "state_changed") {
+      return ``;
+    }
+    if (item.context_event_type! in EVENT_LOCALIZE_MAP) {
+      return `${this.hass.localize(
+        `ui.components.logbook.${EVENT_LOCALIZE_MAP[item.context_event_type!]}`
+      )}`;
+    }
+    return `${this.hass.localize(
+      "ui.components.logbook.from"
+    )} ${this.hass.localize("ui.components.logbook.event")} ${
+      item.context_event_type
+    }`;
+  };
+
+  private _renderEntity(
+    entityId: string | undefined,
+    entityName: string | undefined
+  ) {
+    const hasState = entityId && entityId in this.hass.states;
+    const displayName =
+      entityName ||
+      (hasState
+        ? this.hass.states[entityId].attributes.friendly_name || entityId
+        : entityId);
+    if (!hasState) {
+      return displayName;
+    }
+    return html`<a
+      href="#"
+      @click=${this._entityClicked}
+      .entityId=${entityId}
+      class="name"
+      >${displayName}</a
+    >`;
```

Comment:
We should use a `<button class="link">` for accessibility. 

https://github.com/home-assistant/frontend/blob/dev/src/resources/styles.ts#L151

Reply:
f37e81dee98466dc4a1955e14ce0f068975550eb